client/src/dbus/error.rs

```diff
@@ -13,7 +13,7 @@ pub enum ServiceError {
     /// Session does not exist.
     NoSession,
     /// Collection/Item does not exist.
-    NoSuchObject,
+    NoSuchObject(String),
```

Comment:
This change is wrong I believe. Can you point me to where in gnome-keyring, it passes extra details?

Reply:
https://gitlab.gnome.org/GNOME/gnome-keyring/-/blob/master/daemon/dbus/gkd-secret-service.c#L844

Reply:
Can you check the other errors and update them accordingly as well?

Reply:
Will do.

Reply:
Could you do so as part of the same commit please? To avoid having one commit per enum member.

Reply:
Yeah, look into this now.

Reply:
Yes, IsLocked also takes error message: https://gitlab.gnome.org/GNOME/gnome-keyring/-/blob/master/daemon/dbus/gkd-secret-objects.c#L519. And same for the NoSession: https://gitlab.gnome.org/GNOME/gnome-keyring/-/blob/master/daemon/dbus/gkd-secret-objects.c#L557